fdb-record-layer-lucene/src/test/java/com/apple/foundationdb/record/lucene/LuceneScanAllEntriesTest.java

```diff
@@ -211,4 +226,260 @@ private void assertIndexEntryPrimaryKeyTuples(Set<Tuple> primaryKeys, RecordCurs
+        IndexMaintenanceFilter someFilter = (index, rec) -> {
+            com.google.protobuf.Descriptors.FieldDescriptor recNoField =
+                    rec.getDescriptorForType().findFieldByName("rec_no");
+            if (recNoField != null && rec.hasField(recNoField)) {
+                long recNo = (long) rec.getField(recNoField);
+                if (recNo % 2 == 0) {
+                    // Return SOME to trigger the unsupported operation
+                    return IndexMaintenanceFilter.IndexValues.SOME;
+                }
+            }
+            return IndexMaintenanceFilter.IndexValues.ALL;
+        };
+
+        return getStoreBuilder(context, metaData, path)
+                .setIndexMaintenanceFilter(someFilter);
+    }
+
+    @Nonnull
+    private FDBRecordStore.Builder getStoreBuilderWithFailingFilterFor1002L(@Nonnull FDBRecordContext context,
+                                                                            @Nonnull RecordMetaDataProvider metaData,
+                                                                            @Nonnull final KeySpacePath path) {
+        // Create an index maintenance filter that throws an exception for specific record with recNo 1002L
+        // This can be used to tests error handling when the filter itself fails
+        IndexMaintenanceFilter failingFilter = (index, rec) -> {
+            com.google.protobuf.Descriptors.FieldDescriptor recNoField =
+                    rec.getDescriptorForType().findFieldByName("rec_no");
+            if (recNoField != null && rec.hasField(recNoField)) {
+                long recNo = (long) rec.getField(recNoField);
+                if (recNo == 1002L) {
+                    throw new RecordCoreException("Filter failed for recNo: " + recNo)
+                            .addLogInfo("rec_no", recNo)
+                            .addLogInfo("index", index.getName());
+                }
+                return IndexMaintenanceFilter.IndexValues.ALL;
+            }
+            return IndexMaintenanceFilter.IndexValues.NONE;
+        };
+
+        return getStoreBuilder(context, metaData, path)
+                .setIndexMaintenanceFilter(failingFilter);
+    }
+
+    @ParameterizedTest
+    @BooleanSource
+    public void indexScanWithFailingFilterThrowsExceptionTest(boolean isGrouped) {
+        final long seed = 7654321L;
+
+        final LuceneIndexTestDataModel dataModel =
+                new LuceneIndexTestDataModel.Builder(seed, this::getStoreBuilderWithFailingFilterFor1002L, pathManager)
+                        .setIsGrouped(isGrouped)
+                        .setIsSynthetic(false)
+                        .setPrimaryKeySegmentIndexEnabled(true)
+                        .setPartitionHighWatermark(10)
+                        .build();
+
+        // Attempt to save records - should throw RecordCoreException from the filter for recNo 1002L
+        RecordCoreException exception = Assertions.assertThrows(RecordCoreException.class, () -> {
```

Comment:
This could have a tighter scope around just the `saveRecord` call, right?
In which case, you shouldn't commit.